### PR TITLE
Fix RoomContractSetPriceProcessor and write unit tests

### PR DESCRIPTION
**Pull request by konstyefremov**
Resolves https://github.com/happy-travel/agent-app-project/issues/123

### CODE REVIEW

File: HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/RoomContractSetPriceProcessorTests/ProcessPricesTests.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+        private readonly PriceProcessFunction _priceProcessFunction;
+
+        private readonly RoomContractSet _roomContractSet;
+        private readonly List<RoomContractSet> _roomContractSets;
+
+        private readonly decimal _totalGross;
+        private readonly decimal _totalGrossExpected;
+        
+        private readonly decimal _totalFinal;
```
Comment (kirillta) [inline HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/RoomContractSetPriceProcessorTests/ProcessPricesTests.cs:124]: Do not pass unnecessary variables between methods and code will be clearer

Comment (konstyefremov) [reply HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/RoomContractSetPriceProcessorTests/ProcessPricesTests.cs]: Reworked the test. Reduced the number of variables.

File: HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/RoomContractSetPriceProcessorTests/ProcessPricesTests.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+using HappyTravel.Edo.Api.Services.PriceProcessing;
+using HappyTravel.EdoContracts.General;
+using HappyTravel.Money.Models;
+using Xunit;
+
+namespace HappyTravel.Edo.UnitTests.Tests.Services.Accommodations.Availability.RoomContractSetPriceProcessorTests
+{
+    public class ProcessPricesTests
+    {
+        public ProcessPricesTests()
```
Comment (mingazhev) [inline HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/RoomContractSetPriceProcessorTests/ProcessPricesTests.cs:15]: It is better to make all instantiations and setups in a function called inside a test, please refer to our UnitTests guide

File: HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/RoomContractSetPriceProcessorTests/ProcessPricesTests.cs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            Assert.True(await PriceWasProcessed(_roomContractSets[0].Rate, processed[0].Rate));
+            Assert.True(await PriceWasProcessed(_roomContractSets[0].RoomContracts[0].Rate, processed[0].RoomContracts[0].Rate));
+            Assert.True(await PriceWasProcessed(_roomContractSets[0].RoomContracts[0].DailyRoomRates[0], processed[0].RoomContracts[0].DailyRoomRates[0]));
+        }
+
+
+        private async Task<bool> PriceWasProcessed(Rate original, Rate processed)
+        {
+            return await PriceWasProcessed(original.Gross, processed.Gross) 
+                && await PriceWasProcessed(original.FinalPrice, processed.FinalPrice);
```
Comment (mingazhev) [inline HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/RoomContractSetPriceProcessorTests/ProcessPricesTests.cs:63]: The whole logic of "PriceWasProcessed" is too complex for tests. 
Tests should be as simple as possible (from the logic side) and the best tests are readable without navigating to other functions.
To achieve that it is needed to get the function, calculate desired result manually or with calculator and check against the value that we got from the function.
In this case everyone who sees the test and the real numbers there will see what was initial amount, what was the result, without copying internal logic to the tests logic.
Long story short: we prefer black box testing

Comment (konstyefremov) [reply HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/RoomContractSetPriceProcessorTests/ProcessPricesTests.cs]: Done

File: HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/RoomContractSetPriceProcessorTests/ProcessPricesTests.cs [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+            {
+                new(fromDate,
+                    fromDate.AddDays(1), // because of validation "toDate" should be bigger by exactly 1 day
+                    new MoneyAmount(final, currency),
+                    new MoneyAmount(gross, currency)
+                )
+            };
+        }
+       
+        private readonly PriceProcessFunction AddTenReturnUsd = price => new ValueTask<MoneyAmount>(new MoneyAmount(price.Amount + 10, Currencies.USD)) ;
```
Comment (mingazhev) [inline HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/RoomContractSetPriceProcessorTests/ProcessPricesTests.cs:144]: Maybe it's better to change currency too, but this is optional

**Review by mingazhev — APPROVED** [review]

File: HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/RoomContractSetPriceProcessorTests/ProcessPricesTests.cs (hunk 4 above)
Comment (konstyefremov) [reply HappyTravel.Edo.UnitTests/Tests/Services/Accommodations/Availability/RoomContractSetPriceProcessorTests/ProcessPricesTests.cs:144]: The currency is changed too, actually.
